tests/baselines/reference/computedPropertyNames28_ES5.types

```diff
@@ -12,8 +12,8 @@ class C extends Base {
 >super : typeof Base
 
         var obj = {
->obj : { [x: string]: () => void; }
->{            [(super(), "prop")]() { }        } : { [x: string]: () => void; }
+>obj : { [(super(), "prop")](): void; }

```

Comment:
So, I don't know if it's related to this PR, but this change caught my eye:
```ts
{ [(super(), "prop")](): void; }
```
isn't a valid type. Somewhere we're reusing the name's expression tree when building the computed property name type and shouldn't be. :(

Reply:
This is actually a baseline change from #23592. I somehow messed up a merge and ended up with a lot of apparent changes in this PR.

Anyways, what's not valid about the type? It's a method with a computed property name that uses the comma operator.

Reply:
That's supposed to be type output - if someone wrote `type Foo = { [(super(), "prop")](): void; }`, we'd issue an error.